docs/DesignDocs/AttestationAPIsWithFlexibleClaims.md

```diff
@@ -0,0 +1,476 @@
+Attestation APIs with Init- and Run-Time Claims Support
+==========================
+
+Motivation
+------------
+
+The Open Enclave SDK currently supports attestation APIs,
+`oe_get_attestation_certificate_with_evidence_v2` and
+`oe_verify_attestation_certificate_with_evidence_v2`, that
+allow developers to obtain a self-signed certificate with
+the enclave evidence embedded and to verify the certificate,
+respectively. The latter API also outputs an array of claims,
+allowing the developers to implement additional verification
+logic. However, these APIs have following limitations:
+
+- Lack of support of init-time claims
+  According to [Open Enclave Init-time Configuration Interface](InitTimeConfigurationInterface.md),
+  the support of init-time claims is desirable for the incoming SGX2
+  features (.i.e., `CONFIGID`). The `oe_get_attestation_certificate_with_evidence_v2`
+  API does not allow developers to provide an init-time claim buffer as input.
+
+- Limited support for run-time claims
+  The notion of run-time claims is roughly equivalent to the custom claims
+  in the OE attestation plugin framework. The `oe_get_attestation_certificate_with_evidence_v2`
+  API does not allow developers to provide customized custom claims as input.
+  Instead, the API internally takes the input public key buffer as the custom claim.
+  The `oe_verify_attestation_certificate_with_evidence_v2` then extracts the custom
+  claim and verifies its value against the public key.
+
+Proposal
+--------
+
+This document proposes new APIs,
+`oe_get_attestation_certificate_with_evidence_v3` and
+`oe_verify_attestation_certificate_with_evidence_v3`, that
+address the abovementioned limitations. Note that the certificate
+generated by `oe_get_attestation_certificate_with_evidence_v3`
+does not provide full backward compatibility to older version
+of verification APIs (e.g., `oe_verify_attestation_certificate_with_evidence_v2`
+and `oe_verify_attestation_certificate`).
+
+- oe_get_attestation_certificate_with_evidence_v3
+  ```c
+  /**
+   * oe_get_attestation_certificate_with_evidence_v3
+   *
+   * Similar to oe_get_attestation_certificate_with_evidence_2, this function
+   * generates a self-signed x.509 certificate with embedded evidence generated by
+   * an attester plugin for the enclave, but it also allows a user to pass in
+   * runtime and inittime claims.
+   *
+   * @experimental
+   *
+   * @param[in] format_id The format id of the evidence to be generated.
+   *
+   * @param[in] subject_name a string containing an X.509 distinguished
+   * name (DN) for customizing the generated certificate. This name is also used
+   * as the issuer name because this is a self-signed certificate
+   * See RFC5280 (https://tools.ietf.org/html/rfc5280) for details
+   * Example value "CN=Open Enclave SDK,O=OESDK TLS,C=US"
+   *
+   * @param[in] private_key a private key used to sign this certificate
```

Comment:
nit: half the lines have the param description start with a capital letter, and have with a lower case letter.
Similarly, half end in a period and half don't.  Be consistent.